src/Core/Scripting/OOJSPlayerShip.m

```diff
@@ -644,6 +650,36 @@ static JSBool PlayerShipSetProperty(JSContext *context, JSObject *this, jsid pro
 			}
 			break;
 		
+		case kPlayerShip_chartHightlightMode:
```

Comment:
I don't like this here. It would probably be much better to have this set up in a table in the Core/Tables folder and have OOConstToString.m handle both value to string and string to value conversions. This would allow us to add more modes in the future with only needing to update the table file, rather than adding code here.

What I would try to do is this:
- -Set up the table of LRC modes like below and remove the OOLRC_MODE enum from the headers
```
ENTRY(OOLRC_MODE_UNKNOWN,		-1)
ENTRY(OOLRC_MODE_SUNCOLOR,		0)
ENTRY(OOLRC_MODE_ECONOMY,		1)
ENTRY(OOLRC_MODE_GOVERNMENT,		2)
ENTRY(OOLRC_MODE_TECHLEVEL,		3)
```
- Change the OOStringFromLongRangeChartMode function as seen earlier.
- Add the reverse OOChartModeFromString in OOConstToString.m, defined like the other functions there that take a string and return an integer.
- Rewrite the case statement to maybe something like this:
```
case kPlayerShip_chartHightlightMode:
	sValue = OOStringFromJSValue(context, *value);
	if (sValue) 
	{
		OOLongRangeChartMode chartMode = OOChartModeFromString(sValue);
		if (chartMode > OOLRC_MODE_UNKNOWN)
		{
			[player setLongRangeChartMode:chartMode];
			[player doScriptEvent:OOJSID("chartHighlightModeChanged") withArgument:OOStringFromLongRangeChartMode([player longRangeChartMode])];
			return YES;
		}
		else
		{
			OOJSReportError(context, @"Unknown chart hightlight mode specified - must be either OOLRC_MODE_SUNCOLOR, OOLRC_MODE_ECONOMY, OOLRC_MODE_GOVERNMENT or OOLRC_MODE_TECHLEVEL.");
		}
	}
	return NO;	// not reachable if successfully set
	break;
```
Note that this would rename the CHART_MODE_* labels to OOLRC_MODE_*. Note also that we consider an unknown default mode for the chart, which should be reset to OOLRC_MODE_SUNCOLOR at startup. Note finally that in all the above I have changed OOLRC_MODE_NORMAL to OOLRC_MODE_SUNCOLOR, because I think it is more descriptive.